catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/RemoveCommandTest.java

```diff
@@ -88,7 +114,46 @@ protected SolrCacheMBean getCacheProxy() {
 
         String[] idsArray = new String[ids.size()];
         idsArray = ids.toArray(idsArray);
-        verify(mbean, times(1)).removeById(idsArray);
+        verify(mbeanMock, times(1)).removeById(idsArray);
+    }
+
+    /**
+     * Tests condition where number of results returned by each catalog framework query
+     * is less than batchSize being used.
+     * <p>
+     * Response of size 0 for final call lets the command exit as it simulates
+     * no more results to process.
+     * <p>
+     * Cumulative results by each batch:
+     * First:       19                 (19 results returned)
+     * Second:      37                 (18 results returned)
+     * Third:       54                 (17 results returned)
+     * Fourth:      73                 (19 results returned)
+     * Fifth:       95                 (22 results returned)
+     * Sixth:      100                 (5 results returned)
+     * <p>
+     * Final:       0 results returned, therefore done.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testMultipleCatalogCalls() throws Exception {
```

Comment:
❗️ We probably need to clarify what we're trying to test here and break up this test into multiple ones, or use JUnit's `@Paramterized`.